gateway/src/auth.rs

```diff
@@ -78,17 +79,27 @@ where
             .await
             .map_err(Error::StatusCode)?;
 
-        let scope = match Path::<ProjectName>::from_request_parts(parts, state).await {
-            Ok(Path(p)) => p,
-            Err(_) => Path::<(ProjectName, String)>::from_request_parts(parts, state)
-                .await
-                .map(|Path((p, _))| p)
-                .map_err(|_| InvalidProjectName)?,
-        };
-
         let RouterState { service, .. } = RouterState::from_ref(state);
 
-        let allowed = claim.is_admin()
+        // Enables checking HEAD at /projects/{ulid}, used by res-rec to check permission for a proj id
+        let scope = match Path::<Ulid>::from_request_parts(parts, state).await {
+            Ok(Path(ulid)) => {
+                let p = service.find_project_by_id(&ulid.to_string()).await?;
+                ProjectName::new(&p.name).expect("valid project name")
```

Comment:
Should we mapp to invalidprojectname err instead?

Reply:
Would rather have an internal error than a user facing one here. It should be safe regardless since the name is coming out of the db.